exploit-pocs/exploit-0/exploit-0.ts

```diff
@@ -75,8 +76,9 @@ describe("Exploit 0", async () => {
     exploitSuccessful = false;
 
     let collected_fees = await (await program.account.treasuryLog.fetch(treasury)).collectedFees;
+    const hackerBalance = await provider.connection.getBalance(hacker.publicKey);
 
-    if (collected_fees.toNumber() === 0) {
+    if (collected_fees.toNumber() === 0 && hackerBalance > HACKER_INITIAL_BALANCE) {
```

Comment:
I understand the motivation - the test passes even if the admin withdrew the funds. Your proposed solution works but the hacker keypair has to be used to pass the test. I would rather propose to make the admin keypair as local variable in the setup test.